includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -276,230 +149,140 @@ private function getVerticalData() {
+				'wds-icons-' . $hrefKey,
+				'global-footer-follow-us-link-' . $hrefKey,
+				$hrefUrl,
+				'follow-us.' . $hrefKey
+			);
 		}
 
 		return $data;
 	}
 
 	private function getCommunity() {
 		$data = [
-			'header' => [
-				'type' => 'line-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-header'
-				]
-			],
-			'links' => [ ]
+			'header' => $this->getLineTextTranslatableObject( 'global-footer-community-header' ),
+			'links' => []
 		];
 
-		if ( $this->getHref( 'community-central' ) ) {
-			$data['links'][] = [
-				'type' => 'link-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-link-community-central'
-				],
-				'href' => $this->getHref( 'community-central' ),
-				'tracking_label' => 'community.community-central',
-			];
-		}
-
-		if ( $this->getHref( 'support' ) ) {
-			$data['links'][] = [
-				'type' => 'link-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-link-support'
-				],
-				'href' => $this->getHref( 'support' ),
-				'tracking_label' => 'community.support',
-			];
-		}
-
-		if ( $this->getHref( 'fan-contributor' ) ) {
-			$data['links'][] = [
-				'type' => 'link-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-link-fan-contributor-program'
-				],
-				'href' => $this->getHref( 'fan-contributor' ),
-				'tracking_label' => 'community.fan-contributor',
-			];
-		}
-
-		if ( $this->getHref( 'wam' ) ) {
-			$data['links'][] = [
-				'type' => 'link-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-link-wam-score'
-				],
-				'href' => $this->getHref( 'wam' ),
-				'tracking_label' => 'community.wam',
-			];
-		}
+		$links = [
+			[
+				'titleKey' => 'global-footer-community-link-community-central',
+				'hrefKey' => 'community-central',
+				'trackingLabel' => 'community.community-central'
+			],
+			[
+				'titleKey' => 'global-footer-community-link-support',
+				'hrefKey' => 'support',
+				'trackingLabel' => 'community.support'
+			],
+			[
+				'titleKey' => 'global-footer-community-link-fan-contributor-program',
+				'hrefKey' => 'fan-contributor',
+				'trackingLabel' => 'community.fan-contributor'
+			],
+			[
+				'titleKey' => 'global-footer-community-link-wam-score',
+				'hrefKey' => 'wam',
+				'trackingLabel' => 'community.wam'
+			],
+			[
+				'titleKey' => 'global-footer-community-link-help',
+				'hrefKey' => 'help',
+				'trackingLabel' => 'community.help'
+			],
+		];
 
-		if ( $this->getHref( 'help' ) ) {
-			$data['links'][] = [
-				'type' => 'link-text',
-				'title' => [
-					'type' => 'translatable-text',
-					'key' => 'global-footer-community-link-help'
-				],
-				'href' => $this->getHref( 'help' ),
-				'tracking_label' => 'community.help',
-			];
+		foreach ( $links as $link ) {
```

Comment:
it is a '2 iterations over array vs cleaner code' dilema ;) I don't have strong opinion about this so if you have I can rewrite it.